Sources/BraintreePaymentFlow/BTPaymentFlowClient.swift

```diff
@@ -93,23 +124,41 @@ extension BTPaymentFlowClient: BTPaymentFlowClientDelegate {
             if let error = error as? NSError {
                 if error.domain == ASWebAuthenticationSessionError.errorDomain,
                    error.code == ASWebAuthenticationSessionError.canceledLogin.rawValue {
-                    self._apiClient.sendAnalyticsEvent("ios.\(self.paymentFlowName)).authsession.browser.cancel")
+                    // User canceled by breaking out of the PayPal browser switch flow
+                    // (e.g. System "Cancel" button on permission alert or browser during ASWebAuthenticationSession)
+                    if !self.returnedToAppAfterPermissionAlert {
+                        // User tapped system cancel button on permission alert
+                        self.sendAnalyticsEvent(BTPaymentFlowAnalytics.browserLoginAlertCanceled)
+                    }
+                    self.sendAnalyticsEvent(BTPaymentFlowAnalytics.paymentCanceled)
+                    self.onPaymentComplete(nil, error: BTPaymentFlowError.canceled(self.paymentFlowRequestDelegate?.paymentFlowName() ?? ""))
+                    return
                 }
                 
-                self.onPaymentComplete(nil, error: BTPaymentFlowError.canceled(self.paymentFlowRequestDelegate?.paymentFlowName() ?? ""))
+                self.sendAnalyticsEvent(BTPaymentFlowAnalytics.paymentFailed)
+                self.onPaymentComplete(nil, error: BTPaymentFlowError.webSessionError(error))
                 return
             }
             
             if let callbackURL {
-                self._apiClient.sendAnalyticsEvent("ios.\(self.paymentFlowName).webswitch.succeeded")
                 self.paymentFlowRequestDelegate?.handleOpen(callbackURL)
             } else {
+                self.sendAnalyticsEvent(BTPaymentFlowAnalytics.browserLoginFailed)
+                self.sendAnalyticsEvent(BTPaymentFlowAnalytics.paymentFailed)
                 self.onPaymentComplete(nil, error: BTPaymentFlowError.missingReturnURL)
             }
         })
         
         authenticationSession?.presentationContextProvider = self
-        authenticationSession?.start()
+        isAuthenticationSessionStarted = authenticationSession?.start() ?? false
+        if isAuthenticationSessionStarted {
+            sendAnalyticsEvent(BTPaymentFlowAnalytics.browserPresentationSucceeded)
+        } else {
+            sendAnalyticsEvent(BTPaymentFlowAnalytics.browserPresentationFailed)
+            sendAnalyticsEvent(BTPaymentFlowAnalytics.paymentFailed)
+            onPaymentComplete(nil, error: BTPaymentFlowError.webSessionFailedToLaunch)
+            return
+        }
```

Comment:
This result of this boolean value should only be used for analytics, and should not call the merchant's completion ([see PPCP](https://github.com/paypal/iOS-SDK/blob/3be6c44b1b5fad08d219886cf2d3bfeb7435cf40/Sources/PayPalWebPayments/PayPalWebCheckoutClient.swift#L60-L66)).

Reply:
We followed the same pattern for PayPalClient. https://github.com/braintree/braintree_ios/pull/977/files#diff-7593900bd413ee81d48fae61aeeab899a00cbfdc0d2a688e21d3acf3f7ef3360R358 Isn't failure to launch this authentication page an end event?

Reply:
I've refactored this to return sessionDidDisplay result on successful launch. So we wouldn't be able to send this specific error at that point since we are not returning to merchant app here. I will delete addition of webSessionFailedToLaunch and keep BTPaymentFlowError.webSessionError(error)